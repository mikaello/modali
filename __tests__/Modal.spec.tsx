import React from "react";
import ReactDOM from 'react-dom'
import { create } from "react-test-renderer";
import Modali, { useModali} from "../src/index";

/* Temporary workaround until portals is suported by 'react-test-renderer' */
ReactDOM.createPortal = node => node;

const App = () => {
    const [exampleModal, toggleExampleModal] = useModali();

    return (
        <Modali.Modal isModalVisible hide={() => { }} options={{}}>
        Hi, I'm a Modali!
  </Modali.Modal>)
}

describe("Modali component", () => {
  test("Matches the snapshot", () => {
    const modal = create(<App /> );
    expect(modal.toJSON()).toMatchSnapshot();
  });
});
