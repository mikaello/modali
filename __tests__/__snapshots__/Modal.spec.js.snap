// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modali component Matches the snapshot 1`] = `
Array [
  <div
    className="modali-overlay"
  />,
  <div
    aria-hidden={true}
    aria-modal={true}
    className="modali-wrapper"
    onClick={[Function]}
    role="dialog"
    tabIndex={-1}
  >
    <div
      className="modali modali-size-regular"
    >
      <div
        className="modali-content"
      >
        <div
          className="modali-header"
        >
          <button
            aria-label="Close"
            className="modali-close-button"
            data-dismiss="modal"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
        </div>
        <div
          className="modali-body"
        >
          Hi, I'm a Modali!
        </div>
      </div>
    </div>
  </div>,
]
`;
